packages/ts/file-router/test/vite-plugin/vite-plugin.spec.ts

```diff
@@ -35,54 +96,147 @@ describe('@vaadin/hilla-file-router', () => {
-      expect(mockServer.hot.send).to.be.calledWith({ type: 'custom', event: 'fs-route-update' });
+    it('should send fs-route-update when file-routes.json changes', async () => {
+      expect(mockEnvironment.hot.send).to.not.be.called;
+      const modulesToUpdate = await hotUpdate({ type: 'update', file: fileRoutesJsonPath });
+      expect(mockEnvironment.hot.send).to.be.calledWith({ type: 'custom', event: 'fs-route-update' });
+      expect(modulesToUpdate).to.be.deep.equal([]);
     });
 
-    it('should not send full-reload when other files change', () => {
-      expect(mockServer.hot.send).to.not.be.called;
-      mockServer.watcher.emit('change', fileURLToPath(new URL('file.tsx', viewsDir)));
-      expect(mockServer.hot.send).to.not.be.called;
+    it('should not send updates for file-routes.ts separately', async () => {
+      expect(mockEnvironment.hot.send).to.not.be.called;
+      const modulesToUpdate = await hotUpdate({ type: 'update', file: fileRoutesTsPath });
+      // Updates to file-routes.ts are not processed separately, instead they
+      // are combined with the update of the triggering file.
+      expect(mockEnvironment.hot.send).to.not.be.called;
+      expect(modulesToUpdate).to.be.deep.equal([]);
+    });
+
+    it('should send hot reload including file routes when a view file imported in file routes changes', async () => {
+      expect(mockEnvironment.hot.send).to.not.be.called;
+      const viewFilePath = fileURLToPath(new URL('file.tsx', viewsDir));
+      const viewFileModule = createMockEnvironmentModuleNode(viewFilePath);
+      viewFileModule.importers.add(fileRoutesModule);
+      const modulesToUpdate = await hotUpdate({ type: 'update', file: viewFilePath, modules: [viewFileModule] });
+      expect(mockEnvironment.fetchModule).to.be.calledWith(fileRoutesTsPath);
+      // HMR for routes and view file together.
+      expect(modulesToUpdate).to.not.equal(undefined);
+      expect(modulesToUpdate!.length).to.equal(2);
+      expect(modulesToUpdate![0]).to.eq(fileRoutesModule);
+      expect(modulesToUpdate![1]).to.eq(viewFileModule);
+      expect(mockEnvironment.hot.send).to.not.be.called;
+    });
+
+    it('should send hot reload including file routes when a view file imported in file routes is added', async () => {
+      expect(mockEnvironment.hot.send).to.not.be.called;
+      const viewFilePath = fileURLToPath(new URL('file.tsx', viewsDir));
+      const viewFileModule = createMockEnvironmentModuleNode(viewFilePath);
+      viewFileModule.importers.add(fileRoutesModule);
+      const modulesToUpdate = await hotUpdate({ type: 'create', file: viewFilePath, modules: [viewFileModule] });
+      expect(mockEnvironment.fetchModule).to.be.calledWith(fileRoutesTsPath);
+      // HMR for routes and view file together.
+      expect(modulesToUpdate).to.not.equal(undefined);
+      expect(modulesToUpdate!.length).to.equal(2);
+      expect(modulesToUpdate![0]).to.eq(fileRoutesModule);
```

Comment:
Let's not use that weird `chai`'s aliases 😓 I'm always confused about what they mean. If it is a deep equality, let's have `to.be.deep.equal()`, otherwise `to.be.equal`.